src/vs/workbench/api/common/extHostStatusBar.ts

```diff
@@ -334,10 +351,22 @@ export class ExtHostStatusBar implements ExtHostStatusBarShape {
 		}
 	}
 
+	async $provideTooltip(entryId: string, cancellation: vscode.CancellationToken): Promise<string | IMarkdownString | undefined> {
+		const entry = this._entries.get(entryId);
+		if (!entry) {
+			return undefined;
+		}
+
+		return MarkdownString.fromStrict(await entry.tooltip2?.(cancellation));
+	}
+
 	createStatusBarEntry(extension: IExtensionDescription | undefined, id: string, alignment?: ExtHostStatusBarAlignment, priority?: number): vscode.StatusBarItem;
 	createStatusBarEntry(extension: IExtensionDescription, id?: string, alignment?: ExtHostStatusBarAlignment, priority?: number): vscode.StatusBarItem;
 	createStatusBarEntry(extension: IExtensionDescription, id: string, alignment?: ExtHostStatusBarAlignment, priority?: number): vscode.StatusBarItem {
-		return new ExtHostStatusBarEntry(this._proxy, this._commands, this._existingItems, extension, id, alignment, priority);
+		const entry = new ExtHostStatusBarEntry(this._proxy, this._commands, this._existingItems, extension, id, alignment, priority);
+		this._entries.set(entry.entryId, entry);
```

Comment:
When is the entry removed?

Reply:
As far as I can tell items are only added to `this._existingItems` and `this._entries` and never deleted. The only reason why I had to create a separate Map is because the existing map (`this._existingItems`) contains the `Dto` object and not the `ExtHostStatusBarEntry`. This is where I feel like I am missing context specific to the area so I am looking for your guidance here.

Reply:
A `StatusbarItem` has a `dispose` method, so I think it would be good to clean up then.